includes/misc-functions.php

```diff
@@ -697,6 +697,22 @@ function edd_set_upload_dir( $upload ) {
 }
 
 
+/**
+ * Record an extension error
+ *
+ * A simple wrapper function for edd_record_log()
+ *
+ * @since 2.4
+ * @param string $title Title of the log entry (default: empty)
+ * @param string $message Message to store in the log entry (default: empty)
+ * @param int $parent Parent log entry (default: 0)
+ * @return int ID of the new log entry
+ */
+function edd_record_extension_error( $title = '', $message = '', $parent = 0 ) {
```

Comment:
We should take in a 4th argument for $type, so if an extensions wants they can send in a display ready name of the extension, which could be used to make tabs showing errors for just that extension, and EDD could slug it for them.